subprojects/docs/src/samples/java/jvm-multi-project-with-additional-test-types/groovy/buildSrc/src/main/groovy/myproject.java-conventions.gradle

```diff
@@ -12,14 +12,10 @@ repositories {
 testing {
     suites {
         test {
-            useJUnitPlatform()
-
-            dependencies {
-                implementation 'org.junit.jupiter:junit-jupiter:5.7.1'
-            }
+            useJUnitJupiter()
         }
 
-        integrationTest {
+        integrationTest(JvmTestSuite) {
```

Comment:
I'm curious why we must provide the type argument; is it not implicit when running `suites#create` ?

Reply:
The issue is that the test suite container is a container of `TestSuite`, which is a base class to `JvmTestSuite`.

We don't have an API for specifying a particular subclass of `TestSuite` as the default type. We can work around this by registering two bindings.

`TestSuite` -> `DefaultJvmTestSuite` and `JvmTestSuite` -> `DefaultJvmTestSuite`.

I removed this in https://github.com/gradle/gradle/pull/18308/files#diff-7946b06399ac61873b8952bfb4ab0540afd566fe43c5c8d6ecd89868f4fe6312L58

The problem with this is that elements added without a type can only be assumed to be a `TestSuite` and nothing more specific.

So if you have:
```
suites.withType(JvmTestSuite).all { /* configure Jvm test suite */
suites.register("defaultType")
suites.register("explicitType", JvmTestSuite)
```

The configuration block will fire for `explicitType`, but not for `defaultType` unless `defaultType` is also realized or created eagerly for some other reason. 

This is an issue with our containers because we don't look at the implementation type, only the public type when firing `withType(...)` with non-eager elements. This makes some sense because we don't really know the implementation type until the element is created, but it's surprising.

Reply:
Makes sense. I'm fine with `integrationTest(JvmTestSuite)` then.